test/multiplexed-session.ts

```diff
@@ -0,0 +1,289 @@
+    sandbox.restore();
+  });
+
+  describe('instantiation', () => {
+    it('should correctly initialize the fields', () => {
+      assert.strictEqual(multiplexedSession.database, DATABASE);
+      assert.strictEqual(multiplexedSession.refreshRate, 7);
+      assert.deepStrictEqual(multiplexedSession._multiplexedSession, null);
+      assert(multiplexedSession instanceof events.EventEmitter);
+    });
+  });
+
+  describe('createSession', () => {
+    let _createSessionStub;
+    let _maintainStub;
+
+    beforeEach(() => {
+      _maintainStub = sandbox.stub(multiplexedSession, '_maintain');
+      _createSessionStub = sandbox
+        .stub(multiplexedSession, '_createSession')
+        .resolves();
+    });
+
+    it('should create mux session', () => {
+      multiplexedSession.createSession();
+      assert.strictEqual(_createSessionStub.callCount, 1);
+    });
+
+    it('should start housekeeping', done => {
+      multiplexedSession.createSession();
+      setImmediate(() => {
+        try {
+          assert.strictEqual(_maintainStub.callCount, 1);
+          done();
+        } catch (err) {
+          done(err);
+        }
+      });
+    });
+
+    it('should propagate errors for Multiplexed Session which gets emitted', async () => {
+      const multiplexedSession = new MultiplexedSession(DATABASE);
+      const fakeError = new Error();
+      sandbox.stub(multiplexedSession, '_createSession').rejects(fakeError);
+      const errorPromise = new Promise<void>((resolve, reject) => {
+        multiplexedSession.once('error', err => {
+          if (err === fakeError) {
+            resolve();
+          } else {
+            reject(new Error('Unexpected error emitted'));
+          }
+        });
+      });
+
+      multiplexedSession.createSession();
+
+      await errorPromise;
+    });
+  });
+
+  describe('_maintain', () => {
+    let clock;
+    let createSessionStub;
+    beforeEach(() => {
+      createSessionStub = sandbox
+        .stub(multiplexedSession, '_createSession')
+        .resolves();
+      clock = sandbox.useFakeTimers();
+    });
+    afterEach(() => {
+      clock.restore();
+    });
+    it('should set an interval to refresh mux sessions', () => {
+      const expectedInterval =
+        multiplexedSession.refreshRate! * 24 * 60 * 60000;
+
+      multiplexedSession._maintain();
+      clock.tick(expectedInterval);
+      assert.strictEqual(createSessionStub.callCount, 1);
+    });
+  });
+
+  describe('_createSession', () => {
+    it('should create the mux sessions with multiplexed option', async () => {
+      await multiplexedSession._createSession();
+      assert.strictEqual(createSessionStub.callCount, 1);
+      assert.deepStrictEqual(createSessionStub.lastCall.args[0], {
+        multiplexed: true,
+      });
+    });
+
+    it('should emit the MUX_SESSION_AVAILABLE event on successfully creating mux session', done => {
+      multiplexedSession.on(MUX_SESSION_AVAILABLE, () => {
+        done();
```

Comment:
Add assert like below 
`assert.strictEqual(multiplexedSession._multiplexedSession, "whatever session is stubbed")` 

or maybe check that its not null

Reply:
done